Userland/Libraries/LibWeb/HTML/WebSocket.cpp

```diff
@@ -209,14 +210,30 @@ void WebSocket::on_message(ByteBuffer message, bool is_text)
         return;
     if (is_text) {
         auto text_message = String(ReadonlyBytes(message));
-        MessageEventInit event_init {};
-        event_init.data = JS::js_string(wrapper()->vm(), text_message);
-        event_init.origin = url();
+        MessageEventInit event_init {
+            .data = JS::js_string(wrapper()->vm(), text_message),
+            .origin = url()
+        };
         dispatch_event(MessageEvent::create(EventNames::message, event_init));
         return;
     }
-    // type indicates that the data is Binary and binaryType is "blob"
-    // type indicates that the data is Binary and binaryType is "arraybuffer"
+
+    if (m_binary_type == "blob") {
+        // type indicates that the data is Binary and binaryType is "blob"
+        TODO();
+    } else if (m_binary_type == "arraybuffer") {
+        // type indicates that the data is Binary and binaryType is "arraybuffer"
+        auto& global_obj = wrapper()->global_object();
```

Comment:
[Please do not use abbreviations](https://github.com/SerenityOS/serenity/blob/master/Documentation/CodingStyle.md#wrong). This should be `global_object`.